third_party/blink/renderer/platform/scheduler/common/idle_helper.cc

```diff
@@ -294,12 +294,15 @@ void IdleHelper::OnIdleTaskPostedOnMainThread() {
                "OnIdleTaskPostedOnMainThread");
   if (is_shutdown_)
     return;
-  delegate_->OnPendingTasksChanged(true);
 
   // Avoid updating idle state non-deterministically.
   if (recordreplay::AreEventsDisallowed())
     return;
 
+  // RecordReplay issue RUN-1021
+  // Only call OnPendingTasksChanged when events aren't disallowed.
+  delegate_->OnPendingTasksChanged(true);
```

Comment:
I think, this should go into the `if` below, since only if that branch executes, `PendingTasksChanged` actually happens. In all other cases, the task set does not change - what do you think?

Reply:
Hmm, I don't understand this reasoning --- the upstream code calls OnPendingTasksChanged unconditionally, so we should do so as well.  Essentially we just want the idle helper to ignore tasks that were posted while events are disallowed.

Reply:
Ah - I assumed that you added the `if`. But I just looked back through it and saw that you did not add the `if`, but that diff made it only look like you added it, but indeed it only got moved by your code. That's me not looking at things carefully enough :eyes: